api/src/Data/Core/ContainerClients/ContainerClient.cs

```diff
@@ -53,13 +63,21 @@ public async Task DeleteOneAsync(string id, string partitionKey)
             await this.container.DeleteItemAsync<T>(id, partition);
         }
 
-        private static async Task<IEnumerable<T>> ConstructList(FeedIterator<T> iterator)
+        private static async Task<IDictionary<Guid, T>> ConstructDict(FeedIterator<T> iterator)
```

Comment:
I don't think this correct. CosmosDb doesn't guarantee uniqueness of document Ids. We're only guaranteed uniqueness within a partition which might result in the exception below being thrown.

Reply:
Guids are, by definition, globally unique

Reply:
To elaborate: because Guids are guaranteed to be globally unique identiifers, documents should not contain duplicate Ids, even across partitions. If there are, that's an error.

Reply:
Document Ids need not be Guids. For example, member match records have strings as Ids

Reply:
`T.Id` must be a `Guid` because `T : IModel`

Reply:
T : IModel in ContainerClient. T !: IModel in CosmosDb in general

Reply:
That's fine. This code is only interacting with objects whose Id is a Guid. It's fine as is.

Removing this will just push the problem somewhere else. We need a way to uniquely identify the document a foreign key points to. An alternative is iterating over the IEnumerable doing .First, but that just hides the problem.